packages/app/src/runner/event-manager.ts

```diff
@@ -440,10 +440,23 @@ export class EventManager {
     this.studioStore.setup(config)
 
     const isDefaultProtocolEnabled = Cypress.config('isDefaultProtocolEnabled')
-    const isStudioProtocolEnabled = Cypress.config('isStudioProtocolEnabled')
+
+    let isStudioProtocolEnabled = false
     const isStudioInScope = this.studioStore.isActive || this.studioStore.isLoading
 
-    Cypress.state('isProtocolEnabled', isDefaultProtocolEnabled || (isStudioProtocolEnabled && isStudioInScope))
+    if (isStudioInScope) {
+      await new Promise<void>((resolve) => {
+        this.ws.emit('studio:protocol:enabled', ({ studioProtocolEnabled }) => {
+          isStudioProtocolEnabled = studioProtocolEnabled
+
+          Cypress.state('isProtocolEnabled', isDefaultProtocolEnabled || (isStudioProtocolEnabled && isStudioInScope))
```

Comment:
I think this can be condensed to:

```suggestion
    if (isStudioInScope && !isDefaultProtocolEnabled) {
      await new Promise<void>((resolve) => {
        this.ws.emit('studio:protocol:enabled', ({ studioProtocolEnabled }) => {
          Cypress.state('isProtocolEnabled', studioProtocolEnabled)
```

Reply:
good call, updated

Reply:
good call, updated